gradlePlugin/src/test/java/com/github/spotbugs/SpotBugsPluginTest.java

```diff
@@ -21,65 +23,81 @@
 public class SpotBugsPluginTest extends Assert{
   @Rule
   public TemporaryFolder folder= new TemporaryFolder();
-  
-  @Test
-  public void TestSpotBugsTasksExist() throws Exception{
-    String buildScript = "plugins {\n" + 
-        "  id 'java'\n" +
-        "  id 'com.github.spotbugs'\n" + 
-        "}\n" + 
-        "version = 1.0\n" + 
-        "repositories {\n" + 
-        "  mavenCentral()\n" + 
-        "}";
-    
-    File sourceDir = folder.newFolder("src", "main", "java");
-    File to = new File(sourceDir, "foo.java");
-    File from = new File("src/test/java/com/github/spotbugs/Foo.java");
-    Files.copy(from, to);
-    
-    File buildFile = folder.newFile("build.gradle");
-    Files.write(buildScript.getBytes(), buildFile);
-    
-    
-    BuildResult result = GradleRunner.create().withProjectDir(folder.getRoot()).withArguments(Arrays.asList("tasks", "--all")).withPluginClasspath().build();
-    assertTrue(result.getOutput().contains("spotbugsMain"));
-    assertTrue(result.getOutput().contains("spotbugsTest"));
-  }
 
-  @Test
-  public void testSpotBugsTaskCanRun() throws Exception {
-    String buildScript = "plugins {\n" + 
-        "  id 'java'\n" +
-        "  id 'com.github.spotbugs'\n" + 
-        "}\n" + 
-        "version = 1.0\n" +
-        "repositories {\n" + 
-        "  mavenCentral()\n" + 
-        "  mavenLocal()\n" + 
-        "}";
-    File root = folder.newFolder();
+  /**
+   * The root directory of generated project by {@link #createProject()}
+   */
+  private File root;
```

Comment:
I would use http://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html here instead, since it will automatically clean the folder after each test, thus ensuring no contamination between tests.

Reply:
Nice review! I misunderstood that `TemporaryFolder#getRoot()` returns temporary directory like `/tmp`, but it actually returns `/tmp/junitfoobar`. I will follow your suggestion.